dask/dataframe/groupby.py

```diff
@@ -1110,6 +1112,17 @@ def apply(self, func, *args, **kwargs):
         1.  The user should provide output metadata.
         2.  If the grouper does not align with the index then this causes a full
             shuffle.  The order of rows within each group may not be preserved.
+        3.  Dask's GroupBy.apply is not appropriate for functions reductions.
+            For custom reductions, use :class:`dask.dataframe.Aggregation`.
+
+        .. warning::
+
+           Pandas' groupby-apply can be used to to apply arbitrary functions,
+           including aggregations that result in one row per group. Dask's
+           groupby-apply will apply `func` once to each partition-group pair,
```

Comment:
For code highlighting

```suggestion
           groupby-apply will apply ``func`` once to each partition-group pair,
```